rules/framework/vfs_overlay.bzl

```diff
@@ -37,9 +37,9 @@ def _get_vfs_parent(ctx):
         root_path += ctx.label.workspace_root + "/"
     return root_path + ctx.build_file_path
 
-def _find_top_swiftmodule_file(swiftmodules):
+def _find_swiftmodule_directory(swiftmodules):
     for file in swiftmodules:
-        if file.basename.endswith(".swiftmodule"):
+        if file.basename.endswith(".swiftmodule") and file.is_directory:
```

Comment:
I'm really not familiar with all the edge cases or how the format is supposed to work here. 

From what I understood of the original intent here this was to find the "top-level" swiftmodule file (previously this was either a directory or a source file). The code that uses this then does different things for source vs. directory which this change breaks I believe. 

Perhaps what we can do is:
- Attempt to find any `.swiftmodule` file
- Once found, continue searching
- If we find a `.swiftmodule` directory then we return that
- Otherwise we return the original file we found in step #1

This should keep the original intent while preferring `.swiftmodule` directories which I believe should fix the issue here.

Reply:
```diff
-            swiftmodule_file = _find_top_swiftmodule_file(swiftmodules)
-            if not swiftmodule_file and parent_dir_base_name.endswith(".swiftmodule"):
+            if parent_dir_base_name.endswith(".swiftmodule"):
```

I believe this change makes more sense and I verified it fixes the issue here with the VFS overlay. I've also tested it without the VFS feature in both terminal and via the `xcodeproj` rule. Let me know your thoughts!

Reply:
Ok glad you caught it, the new way seems to make sense so far. This code is very caveated and needs to work end to end with programs which are reading the outputs. I'm going to verify 1-2 edge cases that were left unchecked 👍